src/app/core/services/maprules.service.ts

```diff
@@ -47,39 +48,56 @@ export class MapRulesService {
     }
   }
 
-  isEmpty(empty: any) {
-      return empty === '';
+  saveForm(value: {[name: string]: any}, presetGeometries: any[]){
+    const scrubbedForm = this.serialize(value, presetGeometries);
+    return this.saveNewConfig(scrubbedForm);
   }
 
-  removeEmpty(config: {[name: string]: any}){
-    var next = {}
-    var $scope = this;
-    Object.keys(config).forEach((key) => {
-      if (Array.isArray(config[key])) {
-        var nextArray = [];
-        config[key].forEach(subVal => {
-          if (subVal instanceof Object) {
-            const flush = $scope.removeEmpty(subVal);
-            if (Object.keys(flush).length > 0) {
-              nextArray.push(flush);
+  saveNewConfig(scrubbedForm: {[name: string]: any}){
+    return this.http.post(this.mapRulesUrl, scrubbedForm, this.httpOptions).pipe(
+      catchError(this.handleError)
+    );
+  }
+
+  serialize(config: any, presetGeometries: any[]): any {
+    return {
```

Comment:
This take the form data and turns it into the config.
Reason for having to do this is

1. the form control keys are prefixed w/their placement in the config (so a key of an entry in the primary tags array is `primaryKey`
  - the reason for doing this was that the events fired and handled when combos changed was a bit easier with these names easier with this naming convention
2. I dissociated the geometries checkboxes from the reactive form
3. I managed controls that could have multiple values by saving values to controls as string like `building, house`. So, I needed to do some sort of spliting before we made the save to the api